src/arrayMethodJoin.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinResult = '';
+    let newSeparator = separator;
+
+    if (newSeparator === undefined) {
+      newSeparator = ',';
+    }
+
+    for (const word of this) {
+      switch (word) {
+        case undefined:
+        case null:
+          break;
+
+        default:
+          joinResult += word;
+      }
+
+      if (word !== this[this.length - 1]) {
```

Comment:
This condition may cause incorrect results for repeated elements. Instead, use the loop index to check if it's the last iteration.